src/Leaflet/Core/Plugin.js

```diff
@@ -0,0 +1,14 @@
+exports.onAddRemove_ = function(Nothing, Just, onAdd, onRemove, layer, map) {
+    return function() {
+        var res = {value: Nothing};

```

Comment:
We shouldn't be making assumptions about the representation of refs. We should just do this stuff in PS land.

```purescript
onAddRemove init finish l lf = liftEff do
  ref <- newRef Nothing
  let
    onAddFn = mkFn2 \a b -> do
      val <- init a b
      writeRef ref val
    onRemoveFn = ...
  runFn6 onAddRemove_ ..
  pure ref
```